pkg/kubernetes/osiris_test.go

```diff
@@ -113,3 +113,89 @@ func TestGetMinReplicas(t *testing.T) {
 		})
 	}
 }
+
+func TestGetMetricsCheckInterval(t *testing.T) {
+	testcases := []struct {
+		name           string
+		annotations    map[string]string
+		expectedResult int
+		expectedError  bool
+	}{
+		{
+			name:           "nil map",
+			annotations:    nil,
+			expectedResult: 0,
+			expectedError:  false,
+		},
+		{
+			name:           "empty map",
+			annotations:    map[string]string{},
+			expectedResult: 0,
+			expectedError:  false,
+		},
+		{
+			name: "map with no metrics check interval entry",
+			annotations: map[string]string{
+				"whatever": "60",
+			},
+			expectedResult: 0,
+			expectedError:  false,
+		},
+		{
+			name: "map with invalid metrics check interval entry",
+			annotations: map[string]string{
+				"osiris.deislabs.io/metricsCheckInterval": "invalid",
+			},
+			expectedResult: 0,
+			expectedError:  true,
+		},
+		{
+			name: "map with negative metrics check interval entry",
+			annotations: map[string]string{
+				"osiris.deislabs.io/metricsCheckInterval": "-1",
+			},
+			expectedResult: 0,
+			expectedError:  true,
+		},
+		{
+			name: "map with zero metrics check interval entry",
+			annotations: map[string]string{
+				"osiris.deislabs.io/metricsCheckInterval": "0",
+			},
+			expectedResult: 0,
+			expectedError:  true,
+		},
+		{
+			name: "map with valid metrics check interval entry",
+			annotations: map[string]string{
+				"osiris.deislabs.io/metricsCheckInterval": "60",
+			},
+			expectedResult: 60,
+			expectedError:  false,
+		},
+	}
+
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			actual, err := GetMetricsCheckInterval(test.annotations)
+			if err != nil {
+				if !test.expectedError {
```

Comment:
Would you mind using testify for assertions? I believe that is used fairly consistently across the other unit tests (few though they may admittedly be).

Reply:
done